packages/custom-calendar-builder/src/calendar-builder-app.ts

```diff
@@ -0,0 +1,411 @@
+/**
+ * Calendar Builder ApplicationV2 for creating and editing custom calendars
+ */
+
+export class CalendarBuilderApp extends foundry.applications.api.HandlebarsApplicationMixin(
+  foundry.applications.api.ApplicationV2
+) {
+  private currentJson: string = '';
+  private lastValidationResult: any = null;
+  private coreIntegration: any = null;
+
+  constructor() {
+    super();
+  }
+
+  static DEFAULT_OPTIONS = {
+    id: 'calendar-builder-app',
+    classes: ['calendar-builder'],
+    tag: 'div',
+    window: {
+      frame: true,
+      positioned: true,
+      title: 'CALENDAR_BUILDER.app.title',
+      icon: 'fa-solid fa-edit',
+      minimizable: true,
+      resizable: true,
+    },
+    position: {
+      width: 800,
+      height: 600,
+    },
+    actions: {
+      newCalendar: CalendarBuilderApp.prototype._onNewCalendar,
+      openCalendar: CalendarBuilderApp.prototype._onOpenCalendar,
+      saveCalendar: CalendarBuilderApp.prototype._onSaveCalendar,
+      exportJson: CalendarBuilderApp.prototype._onExportJson,
+      importJson: CalendarBuilderApp.prototype._onImportJson,
+      validateJson: CalendarBuilderApp.prototype._onValidateJson,
+      clearEditor: CalendarBuilderApp.prototype._onClearEditor,
+    },
+  };
+
+  static PARTS = {
+    main: {
+      id: 'main',
+      template: 'modules/seasons-and-stars-calendar-builder/templates/calendar-builder.hbs',
+      scrollable: ['.editor-container', '.validation-results'],
+    },
+  };
```

Comment:
Break the application up into multiple parts. Having the validation results and the editor in the same part will result in the browser losing focus on the editor whenever the validation results change.